code/game/objects/structures/catwalk.dm

```diff
@@ -36,8 +38,12 @@
 	update_icon()
 
 /obj/structure/catwalk/Destroy()
+	var/turf/oldloc = loc
 	redraw_nearby_catwalks()
-	return ..()
+	. = ..()
+	if(istype(oldloc))
+		for(var/atom/movable/AM in oldloc)
+			oldloc.Entered(AM, oldloc)
```

Comment:
There is code that assumes this is only called if `oldloc != src`. I'd just call `fall` on every movable instead, but if there is some reason that's not possible, you'll need to go through `Entered` overrides and insert `oldloc != src` checks, as well as moved event subscribers (possibly; those probably don't make this assumption).

Reply:
fall() turned out to work better than expected.